src/frontend/src/views/Data/index.tsx

```diff
@@ -95,19 +111,24 @@ const Data: FunctionComponent = () => {
       return;
     }
 
-    const transformedData = category.data.map((datum: BioinformaticsData) => {
-      const transformedDatum = Object.assign({}, datum);
-
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- Asserted above
-      category.transforms!.forEach((transform) => {
-        const methodInputs = transform.inputs.map((key) => datum[key]);
-        transformedDatum[transform.key] = transform.method(methodInputs);
-      });
-
-      return transformedDatum;
-    });
+    const transformedData = map(
+      category.data,
+      (datum: BioinformaticsData, key: string) => {
+        const transformedDatum = Object.assign({}, datum);
+
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- Asserted above
+        category.transforms!.forEach((transform) => {
+          const methodInputs = transform.inputs.map(
+            (key: string) => datum[key]
+          );
+          transformedDatum[transform.key] = transform.method(methodInputs);
+        });
+
+        return [key, transformedDatum];
+      }
+    );
 
-    category.data = transformedData as BioinformaticsDataArray;
+    category.data = Object.fromEntries(transformedData);
```

Comment:
Seeing the `Object.fromEntries` here makes me kinda wish the whole id extraction and conversion to lookup dict process (`mapObjectArrayToIdDict`) was smushed together with the transform process. Seems like it would be reasonable to just iterate through the raw data once, doing appropriate transforms to create a prepared data object, then placing that in the lookup dict based on the rules of that data type. Definitely feels like it would be easier to reason about as a single abstracted step that can be pulled into its own configuration function in a separate file. Not really worth it now, but if I do go forward with the table refactor I think that's something I'll play with.